finagle-core/src/main/scala/com/twitter/finagle/Name.scala

```diff
@@ -39,7 +39,11 @@ object Name {
    *
    * Equality of two Names is delegated to `id`. Two Bound instances
    * are equal whenever their `id`s are. `id` identifies the `addr`
-   * and not the `path`.
+   * and not the `path`.  If the `id` is a [[com.twitter.finagle.Name.Path
+   * Path]], it should only contain *bound*--not residual--path components.
```

Comment:
I was hoping *bound* and *unbound* were the operative words and residual was simply an adjective.  Basically, given a path like `/s/buoyant/proxy/s/svc/app/path/to/resource`, a Namer may only bind a prefix like `/s/buoyant/proxy` and leave the residual path, `/s/svc/app/path/to/resource` to be bound later (i.e. by the proxy).  When the proxy binds the remaining name, it may only bind `/s/svc/app`, leaving `/path/to/resource` to be handled by the application in whatever way it deems fit.

I'll try to figure out the best way to fit this into the docs.